src/nn.rs

```diff
@@ -8,18 +8,16 @@ use utils::samples_output_to_matrix;
 
 /// Represents a Neural Network with layers, inputs and outputs
 pub struct NeuralNetwork<T: Activation> {
-    activation: T,
-    layers: Vec<NeuralLayer>,
+    layers: Vec<NeuralLayer<T> >,
     samples: Vec<Sample>,
     error_fn: Option<Box<Fn(f64)>>
 }
 
-impl<T: Activation> NeuralNetwork<T> {
+impl <T: Activation> NeuralNetwork<T> {
```

Comment:
yeah, we need it. like I said, the layers in nn are predicated on a generic type so the vector needs to know what that is which means the nn needs to know what that is.